internal/utils/argocd_client.go

```diff
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	"context"
+	argocdclient "github.com/argoproj/argo-cd/pkg/apiclient"
+	applicationpkg "github.com/argoproj/argo-cd/pkg/apiclient/application"
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"google.golang.org/grpc"
+)
+
+type ArgoCDAppClient interface {
+	Sync(ctx context.Context, in *applicationpkg.ApplicationSyncRequest, opts ...grpc.CallOption) (*v1alpha1.Application, error)
+}
+
+// GetArgoCDAppClient builds an Argo CD application client with the provided configuration
+func GetArgoCDAppClient(c *Configuration) ArgoCDAppClient {
+	acdClientOpts := argocdclient.ClientOptions{
+		ServerAddr: c.ArgoCDServerAddr,
+		Insecure:   true,
```

Comment:
mmmmh I'm not sure about this. could we parameterise this in the Configuration object instead of leaving this insecure by default?